pkg/container-hook/tracer.go

```diff
@@ -154,6 +154,7 @@ var runtimePaths = []string{
 	"/usr/bin/crun",
 	"/usr/bin/conmon",
 	"/var/lib/rancher/k3s/data/current/bin/runc",
+	"/var/lib/rancher/rke2/bin/runc",
```

Comment:
Could you add /var/lib/rancher/k3s/data/current/bin/runc and /var/lib/rancher/rke2/bin/runc in runcfanotify?

https://github.com/inspektor-gadget/inspektor-gadget/blob/f2b9826fc4ae046415cdee30ee4a25322fd3f0c0/pkg/runcfanotify/runcfanotify.go#L121

Reply:
do you also want "/usr/bin/conmon" there?

Reply:
tl;dr: please don't add conmon in the runtimePaths list of package `runcfanotify`.

Package `runcfanotify` does not need to add "conmon" in the `runtimePaths` list because it can read the `conmon` command line in `/proc/$pid/cmdline` when fanotify notices the `crun` execution. `conmon` executes `crun` and the fanotify callback is executed before the execution actually starts, so the `/proc/$pid/cmdline` still refers to the command line of `conmon` (and not yet `crun`). Therefore, we already have all the information in the `crun` fanotify callback. No need for additional fanotify hook on `conmon`.

Package `container-hook` is different: it has been designed to avoid reading `/proc` at all to support `hostPID=false`. It gets the `pid` and the `conmon` command line from ebpf. But getting the command line from ebpf is not as easy as reading `/proc/$pid/cmdline` because the `cmdline` is not in kernel memory but in userspace memory. So it reads the command line from the arguments of the tracepoint sys_enter_execve on `conmon` and we [synchronise](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/docs/devel/fanotify-ebpf.png) with fanotify.

Reply:
While writing the previous comment, it makes me realise:
1. Package container-hook can probably still get the command line from `/host/proc/$pid/cmdline`. That would simplify the ebpf code a lot. We still need the ebpf code to get the pid because the kernel API for fanotify won't give it with `hostPID=false`
2. https://github.com/inspektor-gadget/inspektor-gadget/commit/73acd49caedbf80a1294449211de0da3c1aab947 needs to be applied to [execruntime.bpf.c](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-hook/bpf/execruntime.bpf.c#L36) too. Or alternatively, remove the ebpf code as per 1.

(Not for this PR)